ar_track_alvar/src/kinect_filtering.cpp

```diff
@@ -75,6 +75,18 @@ namespace ar_track_alvar
     return res;
   }
 
+  ARCloud::Ptr projectCloudInPlane(ARCloud::ConstPtr cloud, pcl::ModelCoefficients::ConstPtr plane_coeffs) {
+    ARCloud::Ptr projected_cloud(new ARCloud());
+
+    pcl::ProjectInliers<ARPoint> proj;
+    proj.setModelType(pcl::SACMODEL_PLANE);
+    proj.setInputCloud(cloud);
+    proj.setModelCoefficients(plane_coeffs);
+    proj.filter(*projected_cloud);
+
+    return projected_cloud;
+  }
```

Comment:
I'm wondering whether an orthogonal projection is the best idea? Perhaps an intersection between the plane and the line connecting point and camera origin is a better estimate of the corner position in the plane?
Or maybe, the centroid of the 2d (subpixel?) points and projecting/intersecting the ray of the centroid with the plane?